src/cdk/overlay/position/flexible-connected-position-strategy.ts

```diff
@@ -411,11 +420,38 @@ export class FlexibleConnectedPositionStrategy implements PositionStrategy {
   }
 
   /**
-   * Sets a minimum distance the overlay may be positioned to the edge of the viewport.
+   * Sets a minimum distance the overlay may be positioned from the left edge of the viewport.
+   * @param margin Required margin between the overlay and the viewport edge in pixels.
+   */
+  withViewportMarginLeft(margin: number): this {
```

Comment:
A couple of notes:
1. Instead of introducing properties for each direction, we should just change the viewport margin to be something like `number | {top: number, bottom: number, start: number, end: number}`.
2. We should use `start` and `end`, instead of `left` and `right`, because they get inverted in RTL.

Reply:
Good points, I've updated the parameter to a union type as suggested as well as renamed left/right -> start/end